plugins/rapid7_insightvm/komand_rapid7_insightvm/triggers/new_scans/trigger.py

```diff
@@ -52,36 +50,18 @@ def run(self, params={}):
                     site_scans[site_id] = [max_scan_by_id]
 
             # Open cache to only process previously undiscovered scan IDs
-            try:
-                cache_site_scans = json.loads(util.read_from_cache(self.CACHE_FILE_NAME))
-            except ValueError as e:
-                raise PluginException(cause="Failed to load cache file", assistance=f"Exception returned was {e}")
+            cache_site_scans = NewScans.get_cache_site_scans(self)
 
             # Send scans based on configuration
             for site_id, scans in site_scans.items():
                 for scan in scans:
                     # Only process scan IDs not previously cached
-                    if scan.get("scan_id") in cache_site_scans.get(site_id):
+                    if scan.get("scan_id") in cache_site_scans.get(site_id, []):
                         continue
 
                     # Get scan details
                     endpoint = endpoints.Scan.scans(self.connection.console_url, scan["scan_id"])
-
-                    response = self.connection.session.get(url=endpoint, verify=False)
-                    if response.status_code in [200, 201]:  # 200 is documented, 201 is undocumented
-                        try:
-                            scan_details = response.json()
-                        except json.decoder.JSONDecodeError as e:
-                            raise PluginException(
-                                cause=f"Error: Failed to parse response while retrieving scan "
-                                f"details for scan ID {scan['scan_id']}.",
-                                assistance=f"Exception returned was {e}",
-                            )
-                    else:
-                        raise PluginException(
-                            cause=f"ERROR: Failed to retrieve scan ID {scan['scan_id']}.",
-                            assistance=f"Status code returned was {response.status_code}",
-                        )
+                    scan_details = NewScans.get_scan_details(self, endpoint, scan)
 
                     # Add site name and id; not provided by endpoint
                     scan_details["siteId"] = scan.get("site_id")
```

Comment:
```suggestion
                    scan_details["siteId"] = scan.get("site_id", {})
```

Reply:
Default should be fine here as well as None or {} will be the same outcome.